src/uu/ls/src/ls.rs

```diff
@@ -3019,7 +3020,16 @@ fn display_file_name(
         let absolute_path = fs::canonicalize(&path.p_buf).unwrap_or_default();
         let absolute_path = absolute_path.to_string_lossy();
 
-        // TODO encode path
+        #[cfg(not(target_os = "windows"))]
+        let re = Regex::new(r"[^A-Za-z0-9_\-\.:~/]").unwrap();
+        #[cfg(target_os = "windows")]
+        let re = Regex::new(r"[^A-Za-z0-9_\-\.:~\\]").unwrap();
+
+        // percentage encoding of path
+        let absolute_path = re.replace_all(&absolute_path, |caps: &Captures| -> String {
+            format!("%{:02x}", caps[0].as_bytes().first().unwrap())
+        });
```

Comment:
Would it be possible to do this without a regex by iterating over the characters? Should something happen with the `%` too? According to wikipedia (https://en.wikipedia.org/wiki/Percent-encoding#Percent_character) it should be `%25` I think.

Reply:
It's probably possible without regex, I have to try it. And yes, `%` is encoded as `%25` and it's covered by the test.

Reply:
I removed the regex in the latest push.